ruma-events/src/room/message.rs

```diff
@@ -85,6 +85,337 @@ impl MessageEventContent {
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    /// A contructor to create a html text reply to a message.
+    pub fn text_reply_html(
+        reply: impl Into<String>,
+        html_reply: impl Into<String>,
+        original_message: &MessageEvent,
+    ) -> Self {
+        let quoted = MessageEventContent::get_plain_quote_fallback(original_message);
+        let quoted_html = MessageEventContent::get_html_quote_fallback(original_message);
+
+        let body = format!("{}\n\n{}", quoted, reply.into());
+        let formatted = format!("{}\n\n{}", quoted_html, html_reply.into());
+        let text_message_content =
+            TextMessageEventContent { body, formatted: Some(FormattedBody::html(formatted)) };
+
+        Self {
+            msgtype: MessageType::Text(text_message_content),
+            relates_to: Some(Relation::Reply {
+                in_reply_to: InReplyTo { event_id: original_message.event_id.clone() },
+            }),
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    /// A constructor to create a plain text notice reply to a message.
+    pub fn notice_reply_plain(reply: impl Into<String>, original_message: &MessageEvent) -> Self {
+        let quoted = MessageEventContent::get_plain_quote_fallback(original_message);
+
+        let body = format!("{}\n\n{}", quoted, reply.into());
+
+        let notice_message_content = NoticeMessageEventContent { body, formatted: None };
+
+        Self {
+            msgtype: MessageType::Notice(notice_message_content),
+            relates_to: Some(Relation::Reply {
+                in_reply_to: InReplyTo { event_id: original_message.event_id.clone() },
+            }),
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    /// A constructor to create a html text notice reply to a message.
+    pub fn notice_reply_html(
+        reply: impl Into<String>,
+        html_reply: impl Into<String>,
+        original_message: &MessageEvent,
+    ) -> Self {
+        let quoted = MessageEventContent::get_plain_quote_fallback(original_message);
+        let quoted_html = MessageEventContent::get_html_quote_fallback(original_message);
+
+        let body = format!("{}\n\n{}", quoted, reply.into());
+        let formatted = format!("{}\n\n{}", quoted_html, html_reply.into());
+
+        let notice_message_content =
+            NoticeMessageEventContent { body, formatted: Some(FormattedBody::html(formatted)) };
+
+        Self {
+            msgtype: MessageType::Notice(notice_message_content),
+            relates_to: Some(Relation::Reply {
+                in_reply_to: InReplyTo { event_id: original_message.event_id.clone() },
+            }),
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    fn get_plain_quote_fallback(original_message: &MessageEvent) -> String {
```

Comment:
Is there a particular reason you moved these helper functions into `impl MessageEventContent` (making them associated functions rather than free ones)?

Reply:
There is no real reason, apart from the fact I thought it "made sense" to put them under the same namespace. The last commit makes them free functions again (so that I can easily revert it if it isn't needed for them to be free).
Since they are private functions, would it still be beneficial to add a comment explaining their purpose?